ably.d.ts

```diff
@@ -479,3 +479,29 @@ export declare class Realtime {
   connect: () => void;
   time: (paramsOrCallback?: ablyLib.timeCallback | any, callback?: ablyLib.timeCallback) => void;
 }
+
+export namespace Types {
+  export class ClientOptions {
+    key?: string;
+    clientId?: string;
+
+    /**
+     * When true, time is queried from the Ably servers. Example help text.
+     */
+    queryTime?: boolean;
+
+    constructor(options?: ClientOptions);
+  }
+
+  export class Message {
```

Comment:
I've just realised you've duplicated the Message class, see https://github.com/ably/ably-js/blob/13566592e0c10730f79d797f51e4aaab5deedf91/ably.d.ts#L371-L383

Reply:
Yes, that's necessary if we do not want breaking changes. My plan was not to touch anything in the `ablyLib` namespace, if possible. That means we will have duplicate classes like:

- `ablyLib.Message`
- `Types.Message`

Happy to amend this, if we don't want it.

Reply:
I really don't think we want that.  Is there any evidence that anyone is using explicitly defined types such as `ablyLib.Message` anywhere?  I appreciate this change may break that, but I think that a) the underlying JS will still work so it's not a breaking change, just a breaking typ check change , b) it's going to be very rare that anyone references those types, and if they do, they will simply get a type breaking change which is very easily resolved.

If everyone disagrees, why not set up an alias instead of duplicating types?  I believe:

```typescript
/* within ablyLib */
type Message = Types.Message;
```

would also work